File: Day_4/square_root.cpp
/*
LeetCode Problem 
Problem Statement: Given a non-negative integer x, compute and return the square root of x.

                   Since the return type is an integer, the decimal digits are truncated, and only the integer part of the result is returned.

                   Note: You are not allowed to use any built-in exponent function or operator, such as pow(x, 0.5) or x ** 0.5.
            Input: x = 8
           Output: 2 
*/

#include<bits/stdc++.h>
using namespace std;

int squareRoot(int x){
    if(x==0)
        return 0;
    int start = 1, end = x,res;
    while(true){
        int mid = start + (end-start)/2;
        if(mid > x/mid){
            end = mid-1;
        }
        else{
            if(mid+1>x/(mid+1)){
                res = mid;
                break;
            }
            else
                start = mid+1;
        }
    }
    return res;
}

int main(){
    int x;
    cin>>x;

    cout<<squareRoot(x);
}

File: Day_4/square_root.java
package Day_4;

import java.util.*;

public class square_root{
    public static int squareRoot(int x){
        int start =1, end = x-1,mid,res=0;
        if(x==1)
            res = 1;
        while(start<=end){
            mid = start + (end-start)/2;
            if(mid>x/mid)
                end = mid-1;
            else{
                if(mid+1>x/(mid+1)){
                    res =mid;
                    break;
                }
                else
                    start = mid+1;
            }
        }
        return res;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        System.out.println(squareRoot(x));
        sc.close();
    } 
}